pkg/volume/rbd/rbd.go

```diff
@@ -650,6 +756,69 @@ func (c *rbdUnmounter) TearDownAt(dir string) error {
 	return nil
 }
 
+// <><><>
+var _ volume.BlockVolumeMapper = &rbdDiskMapper{}
+
+type rbdDiskMapper struct {
+	*rbd
+	// capitalized so they can be exported in persistRBD()
+	Mon     []string
+	Id      string
+	Keyring string
+	Secret  string
+	//        fsType        string
+	adminSecret string
+	adminId     string
+	//        mountOptions  []string
+	imageFormat   string
+	imageFeatures []string
+}
+
+// <><><>
+var _ volume.BlockVolumeUnmapper = &rbdDiskUnmapper{}
+
+// GetGlobalMapPath returns global map path and error
+// path: plugins/kubernetes.io/{PluginName}/volumeDevices/{WWID}/{podUid}
+func (rbd *rbdDiskMapper) GetGlobalMapPath(spec *volume.Spec) (string, error) {
+	return rbd.rbdGlobalMapPath(spec)
+}
+
+// GetPodDeviceMapPath returns pod device map path and volume name
+// path: pods/{podUid}/volumeDevices/kubernetes.io~rbd
+// volumeName: pv0001
+func (rbd *rbdDiskMapper) GetPodDeviceMapPath() (string, string) {
+	return rbd.rbdPodDeviceMapPath()
+}
+
+func (rbd *rbdDiskMapper) SetUpDevice() (string, error) {
```

Comment:
rbd plugin has attacher.go, so SetUpDevice() is nothing to do. Just return "" and nil as you implemented here.
Instead, WaitForAttach() is used common attachment method between Filesystem volume and Block volume.
You need to update attacher logic to handle both.